sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -199,27 +208,53 @@ export class SenderImpl implements Sender {
     return this._isClosed || this._context.isClosed;
   }
 
-  async send(message: ServiceBusMessage, options?: OperationOptions): Promise<void> {
+  async send(message: ServiceBusMessage, options?: OperationOptions): Promise<void>;
+  async send(messages: ServiceBusMessage[], options?: OperationOptions): Promise<void>;
+  async send(messageBatch: ServiceBusMessageBatch, options?: OperationOptions): Promise<void>;
+  async send(
+    messageOrMessagesOrBatch: ServiceBusMessage | ServiceBusMessage[] | ServiceBusMessageBatch,
+    options?: OperationOptions
+  ): Promise<void> {
     this._throwIfSenderOrConnectionClosed();
-    throwTypeErrorIfParameterMissing(this._context.namespace.connectionId, "message", message);
-    return this._sender.send(message, options);
+
+    if (Array.isArray(messageOrMessagesOrBatch)) {
+      const batch = await this.createBatch(options);
+
+      for (const message of messageOrMessagesOrBatch) {
+        if (!batch.tryAdd(message)) {
+          // this is too big - throw an error
+          const error = new MessagingError(
+            "Messages were too big to fit in a single batch. Remove some messages and try again or create your own batch using createBatch(), which gives more fine-grained control."
+          );
+          error.code = "MessageTooLargeError";
+          throw error;
+        }
+      }
+
+      return this.send(batch, options);
```

Comment:
why not use `this._sender.sendBatch` here?

Reply:
Generally I try to stick with using the public interface, when I can.

Reply:
Given that other if/else sections are doing some checks, some pre-work and then passing off the work to the internal sender, I would expect the same pattern to be used here as well. 

When reading the code here, for a second I panicked thinking, "but this is a batch, we need to send it to the sendBatch() method`, but later realizing that we are not working with the internal sender here

Reply:
This also results in a series of calls that can be avoided, one to the public send, another check on whether sender/conenction is closed, then check to see if it is an array, then check to see if it is a batch, and only then send the thing vs just send the thing

Reply:
Very good points, changed it over since my test run failed :)